framework/components/s3provider/minio.go

```diff
@@ -0,0 +1,241 @@
+package s3provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/go-connections/nat"
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"github.com/smartcontractkit/chainlink-testing-framework/framework"
+	tc "github.com/testcontainers/testcontainers-go"
+	tcwait "github.com/testcontainers/testcontainers-go/wait"
+	"math/rand"
+	"os"
+	"strconv"
+)
+
+const (
+	DefaultImage       = "minio/minio"
+	DefaultName        = "minio"
+	DefaultBucket      = "test-bucket"
+	DefaultRegion      = "us-east-1"
+	DefaultPort        = 9000
+	DefaultConsolePort = 9001
+)
+
+type Minio struct {
+	host        string
+	port        int
+	consolePort int
+	accessKey   string
+	secretKey   string
+	bucket      string
+	region      string
+	keep        bool
+}
+
+func (m Minio) GetSecretKey() string {
+	return m.secretKey
+}
+
+func (m Minio) GetAccessKey() string {
+	return m.accessKey
+}
+
+func (m Minio) GetBucket() string {
+	return m.bucket
+}
+
+func (m Minio) GetConsoleURL() string {
+	return fmt.Sprintf("http://%s:%d", m.host, m.consolePort)
+}
+
+func (m Minio) GetEndpoint() string {
+	return fmt.Sprintf("%s:%d", m.host, m.port)
+}
+
+func (m Minio) GetRegion() string {
+	return m.region
+}
+
+type Option func(*Minio)
+
+type MinioFactory struct{}
+
+func NewMinioFactory() ProviderFactory {
+	return MinioFactory{}
+}
+
+func (mf MinioFactory) New(options ...Option) (Provider, error) {
+	m := &Minio{
+		port:        DefaultPort,
+		consolePort: DefaultConsolePort,
+		accessKey:   randomStr(20),
```

Comment:
It's a local testing environment, do we need `accessKey` and `secretKey` to be random, or can we keep them as constants?

Reply:
how can user overwrite it? programmatically when creating a new instance?